base/set.jl

```diff
@@ -103,6 +116,30 @@ const ⊆ = issubset
 ⊊(l::Set, r::Set) = <(l, r)
 ⊈(l::Set, r::Set) = !⊆(l, r)
 
+doc"""
+    unique(C,k::Integer)
+
+Returns up to k distinct values in C in the order they appear
```

Comment:
@tkelman - yes there was missing coverage for 1x1 cases on a few methods, as well as some 2x2 special cases that I hadn't considered at all before. Will fix it. 

@nalimilan @hayd - I `collect` on associative types because that is the current behavior and I didn't want to break any user code that might depend on it. I think this makes sense, in that even though associative types don't have an ordering, their iterators do. It might be worht emphasizing this in the doc for unique.